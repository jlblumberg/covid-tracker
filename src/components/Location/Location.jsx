import React, { useState, useEffect } from 'react';
import { NativeSelect, FormControl } from '@material-ui/core';
import styles from './Location.module.css';
import { fetchCountries } from '../../api/index.js'


const Location = ({ handleCountryChange }) => {
  const [fetchedCountries, setFetchedCountries] = useState([]);

  useEffect(() => {
    const loadCountries = async () => {
      const countries = await fetchCountries();
      setFetchedCountries(countries)
    }

    loadCountries();
  }, [setFetchedCountries]);

  const formatData = () => {
    if (fetchedCountries.constructor === Array) {
      return fetchedCountries.map((country, index) => <option key={index} value={country}>{country}</option>)
    }
  }

  return(
    <FormControl className={styles.formControl} id='location-picker'>
      <NativeSelect defaultValue='' id='location-options' onChange={(e) => handleCountryChange(e.target.value)}>
        <option value='global'>Global</option>
        {formatData()}
      </NativeSelect>
    </FormControl>
  )

}

export default Location;