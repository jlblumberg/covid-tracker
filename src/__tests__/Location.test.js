import React from 'react';
import { shallow } from 'enzyme';
import Location from '../components/Location/Location';

describe('Location', () => {
  
  it('renders correctly', () => {
    const wrapper = shallow(<Location/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('has a form to choose the location to plot', () => {
    const wrapper = shallow(<Location/>);
    expect(wrapper.find('#location-picker').exists()).toEqual(true);
  });

});

