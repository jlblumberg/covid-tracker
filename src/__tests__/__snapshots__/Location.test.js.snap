// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location renders correctly 1`] = `ShallowWrapper {}`;
